File: internal/session/session.go
package session

import (
	"fmt"
	"os"
	"path/filepath"
)

// StoragePath returns the default Wick data directory.
func StoragePath() (string, error) {
	home, err := os.UserHomeDir()
	if err != nil {
		return "", fmt.Errorf("cannot determine home directory: %w", err)
	}
	return filepath.Join(home, ".wick", "data"), nil
}

// ClearSession removes all persistent data (cookies, cache) and
// recreates the directory.
func ClearSession() error {
	path, err := StoragePath()
	if err != nil {
		return err
	}
	if err := os.RemoveAll(path); err != nil {
		return err
	}
	return os.MkdirAll(path, 0700)
}

File: rust/src/session.rs
use anyhow::Result;
use std::path::PathBuf;

pub fn storage_path() -> Result<PathBuf> {
    let home = dirs_next().ok_or_else(|| anyhow::anyhow!("cannot determine home directory"))?;
    Ok(home.join(".wick").join("data"))
}

pub fn clear() -> Result<()> {
    let path = storage_path()?;
    if path.exists() {
        std::fs::remove_dir_all(&path)?;
    }
    std::fs::create_dir_all(&path)?;
    Ok(())
}

fn dirs_next() -> Option<PathBuf> {
    std::env::var_os("HOME").map(PathBuf::from)
}
